zerver/openapi/test_curl_examples.py

```diff
@@ -50,9 +50,14 @@ def test_generated_curl_examples_for_success(client: Client) -> None:
                 unescaped_html = html.unescape(curl_command_html)
                 curl_command_text = unescaped_html[len("<p><code>curl\n") : -len("</code></p>")]
 
+                reactivation_flag = 0
+                user_id = 0
                 curl_command_text = curl_command_text.replace(
                     "BOT_EMAIL_ADDRESS:BOT_API_KEY", authentication_line
                 )
+                if file_name == "templates/zerver/api/deactivate-own-user.md":
+                    reactivation_flag = 1
+                    user_id = client.get_profile()["user_id"]
```

Comment:
This is super hacky -- `reactivation_flag` should be a bool if we were going to do this, but we have tried hard to keep per-endpoint code out of this function, and I'd like to avoid regressing that.  E.g.

```
diff --git a/zerver/openapi/test_curl_examples.py b/zerver/openapi/test_curl_examples.py
index ba4f431fe1..72a0f9b43e 100644
--- a/zerver/openapi/test_curl_examples.py
+++ b/zerver/openapi/test_curl_examples.py
@@ -49,14 +49,20 @@ def test_generated_curl_examples_for_success(client: Client, owner_client: Clien
                 curl_command_html = md_engine.convert(line.strip())
                 unescaped_html = html.unescape(curl_command_html)
                 curl_command_text = unescaped_html[len("<p><code>curl\n") : -len("</code></p>")]
-
-                reactivation_flag = 0
-                user_id = 0
                 curl_command_text = curl_command_text.replace(
                     "BOT_EMAIL_ADDRESS:BOT_API_KEY", authentication_line
                 )
+
+                # TODO: This needs_reactivation block is a hack.
+                # However, it's awkward to test the "deactivate
+                # myself" endpoint with how this system tries to use
+                # the same account for all tests without some special
+                # logic for that endpoint; and the hack is better than
+                # just not documenting the endpoint.
+                needs_reactivation = False
+                user_id = 0
                 if file_name == "templates/zerver/api/deactivate-own-user.md":
-                    reactivation_flag = 1
+                    needs_reactivation = True
                     user_id = client.get_profile()["user_id"]
 
                 print("Testing {} ...".format(curl_command_text.split("\n")[0]))
@@ -74,7 +80,7 @@ def test_generated_curl_examples_for_success(client: Client, owner_client: Clien
                     )
                     response = json.loads(response_json)
                     assert response["result"] == "success"
-                    if reactivation_flag:
+                    if needs_reactivation:
                         owner_client.reactivate_user_by_id(user_id)
                 except (AssertionError, Exception):
                     error_template = """
```

The right approach is to define something like the existing one for deactivating a user:
```
@openapi_param_value_generator(["/users/{user_id}:delete"])                              
def deactivate_user() -> Dict[str, object]:                                              
    user_profile = do_create_user(                                                       
        email="testuser@zulip.com",                                                      
        password=None,                                                                   
        full_name="test_user",                                                           
        realm=get_realm("zulip"),                                                        
        acting_user=None,                                                                
    )                                                                                    
    return {"user_id": user_profile.id}                                                  
```

Namely, we should just create a new user and then have that user deactivate itself, if we can make that work.

Hmm, except that's not going to work easily without more refactoring, because this test framework doesn't support using a different user account.  OK, I think I'm on board with leaving the work of trying to do something better here for a follow-up issue -- can you open a follow-up issue to track the need to replace this with something less hacky?